CMakeLists.txt

```diff
@@ -38,11 +38,11 @@ file(READ version.config VERSION_CONFIG)
 
 if(VERSION_CONFIG
    MATCHES
-   "(^|.*[^a-z])version[\t ]*=[\t ]*([0-9]+\\.[0-9]+\\.*[0-9]*)-([a-z]+[0-9]*|dev)?.*"
+   "(^|.*[^a-z])version[\t ]*=[\t ]*([0-9]+\\.[0-9]+\\.*[0-9]*)(-([a-z]+[0-9]*|dev))?.*"
 )
   set(VERSION ${CMAKE_MATCH_2})
   if(CMAKE_MATCH_3)
-    set(PROJECT_VERSION_MOD ${CMAKE_MATCH_2}-${CMAKE_MATCH_3})
+    set(PROJECT_VERSION_MOD ${CMAKE_MATCH_2}${CMAKE_MATCH_3})
```

Comment:
I am not sure this is correct. The intention previously, I think, was for `${CMAKE_MATCH_3}` to contain the suffix without the dash, now it will include the suffix. Note that the value captured here is later used to set `${VERSION_MOD}`, which should not include the dash.

Reply:
`VERSION_MOD` was only used to set `PROJECT_VERSION_MOD`, so it is not needed any more.

Reply:
Hmm, I guess things have changed significantly in `master` since I last looked. Was this issue introduced recently?

Reply:
Ok, I guess I need to take a deeper look because a lot of things has changed that aren't obvious from this PR alone.

Reply:
I refactored code in https://github.com/timescale/timescaledb/pull/3381 to be able to keep the update and downgrade version separately. It is needed for the release procedure to work.

Reply:
So, previously, we had separate variables for MAJOR, MINOR, PATCH and SUFFIX (mod version). But if this has been refactored and we no longer care about that, then I guess it is fine.

Reply:
It's even more complicated than that. We had one variable `VERSION_MOD` that contained suffixes like `-rc1` and one variable `VERSION_DEV` that contained the suffix `-dev`. This was then used to build the `PROJECT_VERSION_MOD`. We didn't really have anything like `MAJOR`, `MINOR`, and `PATCH` variables, which would have been natural.